build-scripts/build_js_compiler.js

```diff
@@ -0,0 +1,147 @@
+ * In this case the compiler will be built as a SNAPSHOT.
+ */
+'use strict';
+
+const ncp = require('ncp');
+const fs = require('fs');
+const path = require('path');
+const runCommand = require('./run-command');
+
+/**
+ * Retrieves the compiler version that will be built by reading the contents of ./compiler/pom.xml.
+ * For release builds, this is of the form: "vYYYYMMDD"
+ * For nightly builds, this is "1.0-SNAPSHOT"
+ *
+ * @return {string}
+ */
+function getCompilerVersionFromPomXml() {
+  const pomXmlContents = fs.readFileSync(path.resolve(__dirname, '..', 'compiler', 'pom.xml'), 'utf8');
+  const versionParts = /<version>([^<]+)<\/version>/.exec(pomXmlContents);
+  return versionParts[1];
+}
+
+let compilerVersion = getCompilerVersionFromPomXml();
+const compilerJavaBinaryPath = `./compiler/target/closure-compiler-${compilerVersion}.jar`;
+const compilerJsBinaryPath = `./compiler/target/closure-compiler-gwt-${compilerVersion}/jscomp/jscomp.js`;
+
+console.log(process.platform, process.arch, compilerVersion);
+
+/**
+ * @param {string} src path to source file or folder
+ * @param {string} dest path to destination file or folder
+ * @return {!Promise<undefined>}
+ */
+function copy(src, dest) {
+  return new Promise((resolve, reject) => {
+    ncp(src, dest, err => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+/**
+ * Copy the newly built compiler and the contrib folder to the applicable packages.
+ *
+ * @return {!Promise<undefined>}
+ */
+function copyCompilerBinaries() {
+  return Promise.all([
+    copy(compilerJsBinaryPath, './packages/google-closure-compiler-js/jscomp.js'),
+  ]);
+}
+
+const mvnCmd = `mvn${process.platform === 'win32' ? '.cmd' : ''}`;
+
+if (!fs.existsSync(compilerJavaBinaryPath) || !fs.existsSync(compilerJsBinaryPath)) {
+  // Force maven to use colorized output
+  const extraMvnArgs = process.env.GITHUB_ACTIONS ? ['-Dstyle.color=always'] : [];
+  if ((process.env.GITHUB_ACTIONS)) {
+    process.env.MAVEN_OPTS = '-Djansi.force=true';
+  }
+
+  runCommand(mvnCmd, extraMvnArgs.concat(['clean']), {cwd: './compiler'})
+      .then(({exitCode}) => {
+        if (exitCode !== 0) {
+          process.exit(exitCode);
+          return;
+        }
+        return runCommand(
+            mvnCmd,
+            extraMvnArgs.concat([
+              '-DskipTests',
+              '-pl',
+              'externs/pom.xml,pom-gwt.xml',
+              'install'
+            ]),
+            {cwd: './compiler'});
+      })
+      .then(({exitCode}) => {
+        if (exitCode !== 0) {
+          process.exit(exitCode);
+          return;
+        }
+        // Add a license header to the gwt version jscomp.js file since the compiler build omits this.
+        // If the gwt version ever has a source map, the source mappings will need updated to account for the
+        // prepended lines.
+        const jscompFileContents = fs.readFileSync(compilerJsBinaryPath, 'utf8');
```

Comment:
Are you intending to land the PR for removing the JS-compiled version of closure-compiler before this PR?

Reply:
I think I'd prefer to land this PR before removing the JS version. The number of different systems (Travis + Appveyeor + Github + NPM) and their interactions is incredibly complicated. It's much simplified after this.

Reply:
Actually no I can't. Building the windows native image on Java 8 can only be done with a very very old windows SDK (windows 7). Appveyor is the only build system I could find to support that - and it's extremely difficult to maintain.

GitHub actions allowed updating to java 11 for the main build, but java 8 for GWT. This massively simplified the build.